docs/vault/guide.md

```diff
@@ -118,36 +139,92 @@ Start the Vault:
 
-</details>
 
 ## Usage
 
 ### Connecting the Vault to Rococo
+Connect to our PolkaBTC node or run your own, as descriebd [above](#_5-start-the-vault-client).
+
+Run the vault
+
+```sh
+./vault \
+  --bitcoin-rpc-url http://localhost:18332 \
+  --bitcoin-rpc-user rpcuser \
+  --bitcoin-rpc-pass rpcpass \
+  --keyfile keyfile.json \
+  --keyname myvault \
+  --polka-btc-url 'wss://rococo.polkabtc.io/api/parachain'
+```
 
 ### Registering your Vault
+To automatically register the vault when starting it, use the `auto-register-with-collateral` or `auto-register-with-faucet-url` flags, as described in the [README](https://github.com/interlay/polkabtc-clients/tree/master/vault).
+
+You can also register your vault through our web UI, going to the "Vault" tab, clicking the `Register` button and completing the steps.
 
+Moreover, you can interact with the Vault client directly using [polkabtc-js](https://github.com/interlay/polkabtc-js).
+```js
+import { VaultClient } from "@interlay/polkabtc";
+const vaultClient = new VaultClient(VAULT_CLIENT_URL);
+
+// 100 DOT denominated in Planck
+const collateralInPlanck = "1000000000000";
+await vaultClient.registerVault(collateralInPlanck);
+```
 ### Increasing Collateral
+**Web UI**
+
+Go to the Vault tab and click on button next to the `Collateral:   X DOT for Y BTC` text (above the Issue Requests table). Then, follow the isntructions.
 
+**Polkabtc-js library**
+
+You can interact with the Vault directly client using [polkabtc-js](https://github.com/interlay/polkabtc-js).
+```js
+import { VaultClient } from "@interlay/polkabtc";
+const vaultClient = new VaultClient(VAULT_CLIENT_URL);
+
+// 100 DOT denominated in Planck
+const additionalCollateralInPlanck = "1000000000000";
+await vaultClient.lockAdditionalCollateral(additionalCollateralInPlanck);
+```
 ### Withdrawing Collateral
+**Web UI**
+
+Go to the Vault tab and click on the button next to the `Collateral:   X DOT for Y BTC` text (above the Issue Requests table). Then, follow the isntructions.
+
+**Polkabtc-js library**
+
+You can interact with the Vault directly client using [polkabtc-js](https://github.com/interlay/polkabtc-js).
+```js
+import { VaultClient } from "@interlay/polkabtc";
+const vaultClient = new VaultClient(VAULT_CLIENT_URL);
+
+// 100 DOT denominated in Planck
+const collateralToWithdrawInPlanck = "1000000000000";
+await vaultClient.withdrawCollateral(collateralToWithdrawInPlanck);
+```
 
 ### Earning Fees
+See the Fee Model described in the Overview section.
 
 ### Accepting Issue and Redeem Requests
+Issue and Redeem requests are processed automatically at the moment, signing transactions on the Bitcoin and Polkadot networks using the mnemonic/account credentials you provide to the client when running it.
 
 ### Leaving PolkaBTC
+Even if a vault can withdraw all of its collateral and go offline, it cannot call a function to deregister from PolkaBTC.
```

Comment:
Change this to:
The process to leave PolkaBTC depends on whether or not your Vault client holds BTC in custody.

If you Vault has *no BTC in custody*, you can withdraw all your DOT collateral at any time and leave the system. It is safe to stop the Vault client woithout risking being penalized. You will not participate in any issue or redeem requests once you have removed your DOT collateral.

If your Vault clients holds at least *some BTC in custody*, you have two options to leave the system. Both options require that the BTC that you have in custody is moved. Option A, leaving through *replace*, requires you to request being replaced by another vault. You can request to be replaced through the Vault dashboard [ADD LINK]. Option B, leaving through *redeem* requires you to wait for a user to redeem the entire amount of BTC that the Vault has in custody. Only after you have 0 BTC, can the Vault client withdraw its entire collateral.